context/src/main/java/io/opentelemetry/context/internal/shaded/AbstractWeakConcurrentMap.java

```diff
@@ -283,7 +283,7 @@ public boolean equals(Object other) {
       if (other instanceof WeakKey<?>) {
         return ((WeakKey<?>) other).get() == get();
       } else {
-        return other.equals(this);
+        return other == null ? false : other.equals(this);
```

Comment:
```suggestion
        return other != null && other.equals(this);
```

I prefer not using `?:` in purely boolean expressions.